skorch/setter.py

```diff
@@ -0,0 +1,42 @@
+"""Setter functions for virtual params such as ``optimizer__lr``."""
+import re
+
+from skorch.utils import set_optimizer_param
+
+
+def optimizer_setter(
+        net, param, value, optimizer_attr='optimizer_', optimizer_name='optimizer'
+    ):
+    """Handle setting of optimizer parameters such as learning rate and
+    parameter group specific parameters such as momentum.
+    """
+    if param == 'lr':
+        param_group = 'all'
+        param_name = 'lr'
+        net.lr = value
+    else:
+        pat_1 = '__param_groups__(?P<group>[0-9])__(?P<name>.+)'
```

Comment:
I find this whole section a little bit complicated. Perhaps it would already help if it is moved to a sub-function with a good name.